mir_eval/separation.py

```diff
@@ -474,26 +702,40 @@ def evaluate(reference_sources, estimated_sources, **kwargs):
     # Compute all the metrics
     scores = collections.OrderedDict()
 
-    sdr, sir, sar, perm = util.filter_kwargs(
-        bss_eval_sources_framewise,
+    sdr, isr, sir, sar, perm = util.filter_kwargs(
+        bss_eval_images,
         reference_sources,
         estimated_sources,
         **kwargs
     )
-    scores['Sources Frames - Source to Distortion'] = sdr.tolist()
-    scores['Sources Frames - Source to Interference'] = sir.tolist()
-    scores['Sources Frames - Source to Artifact'] = sar.tolist()
-    scores['Sources Frames - Source permutation'] = perm
-
-    sdr, sir, sar, perm = util.filter_kwargs(
-        bss_eval_sources,
-        reference_sources,
-        estimated_sources,
-        **kwargs
-    )
-    scores['Sources - Source to Distortion'] = sdr.tolist()
-    scores['Sources - Source to Interference'] = sir.tolist()
-    scores['Sources - Source to Artifact'] = sar.tolist()
-    scores['Sources - Source permutation'] = perm
+    scores['Images - Source to Distortion'] = sdr.tolist()
+    scores['Images - Image to Spatial'] = isr.tolist()
+    scores['Images - Source to Interference'] = sir.tolist()
+    scores['Images - Source to Artifact'] = sar.tolist()
+    scores['Images - Source permutation'] = perm.tolist()
+
+    # Verify we can compute sources on this input
+    if reference_sources.ndim < 3 and estimated_sources.ndim < 3:
```

Comment:
It's probably worth making a note in the docstring about how sources will only be evaluated when the nchan dimension is not provided.